test/zdtm.py

```diff
@@ -1491,6 +1492,7 @@ def restore(self):
             pstree_check_stopped(self.__test.getpid())
             pstree_signal(self.__test.getpid(), signal.SIGCONT)
 
+    @functools.cache
```

Comment:
https://github.com/checkpoint-restore/criu/actions/runs/5286913168/jobs/9566809091?pr=2198

```
+ ./test/zdtm.py run -a -p 2 --keep-going
3032
Traceback (most recent call last):
3033
  File "./test/zdtm.py", line 1046, in <module>
3034
    class criu:
3035
  File "./test/zdtm.py", line 1501, in criu
3036
    @functools.cache
3037
AttributeError: module 'functools' has no attribute 'cache'
```

Reply:
pls add some numbers to show how much faster it is with this change.

Reply:
`functools.cache` is Python 3.9+ [1]. I'll drop this patch for now.

[1] https://docs.python.org/3/library/functools.html#functools.cache